joern-cli/frontends/rubysrc2cpg/src/main/antlr4/io/joern/rubysrc2cpg/parser/RubyParser.g4

```diff
@@ -216,10 +216,12 @@ argumentList
         # operatorsArgumentList
     |   associationList (COMMA NL* splattingArgument)? (COMMA NL* blockArgument)?
         # associationsArgumentList
+    |   LBRACK ((symbol|association) COMMA? NL*)* RBRACK
```

Comment:
The following change also works. I see that `[1, 2, 3]` would work as an argument, and that the issue was the symbols/associations as this parser rules implies. I then followed why normal arrays weren't handling it and found there wasn't a rule for these in "indexingArgumentList"s. 

Perhaps my proposed solution will be a bit more general? As it covers array bodies elsewhere too

![Screenshot 2024-08-28 at 08 18 04](https://github.com/user-attachments/assets/aaa8a88b-c466-4966-a08c-1655425b1852)

Reply:
Might want to name it `symbolOrAssocationIndexingArgumentList`